clients/src/main/java/org/apache/kafka/common/requests/ProduceResponse.java

```diff
@@ -222,7 +222,8 @@ public ProduceResponse(Struct struct) {
                 int partition = partRespStruct.get(PARTITION_ID);
                 Errors error = Errors.forCode(partRespStruct.get(ERROR_CODE));
                 long offset = partRespStruct.getLong(BASE_OFFSET_KEY_NAME);
-                long logAppendTime = partRespStruct.getLong(LOG_APPEND_TIME_KEY_NAME);
+                long logAppendTime = partRespStruct.hasField(LOG_APPEND_TIME_KEY_NAME) ?
```

Comment:
Should we convert the type of LOG_APPEND_TIME_KEY_NAME from long to Field.Int64? the benefit of using Field.Int64 is shown below.
1. remove the duplicate code of converting LOG_APPEND_TIME_KEY_NAME  to Field.Int64
1. use more readable method - getOrElse

Reply:
@chia7712 Thanks for the review. As @hachikuji mentioned, we can convert to use the generated classes, so will leave that for a separate PR.